ner_v2/detectors/temporal/date/date_detection.py

```diff
@@ -209,6 +223,14 @@ def _detect_range(self):
             date_dict_list[0][detector_constant.DATE_START_RANGE_PROPERTY] = True
             date_dict_list[-1][detector_constant.DATE_END_RANGE_PROPERTY] = True
 
+        if patterns_2:
+            for pattern in patterns_2:
```

Comment:
We should be stricter here for lines 227 - 232

```suggestion
            for pattern in patterns_2:
                date_dicts =  self._date_dict_from_text(text=pattern[0])
                if len(date_dicts) == 2:
                    date_dicts[0][detector_constant.DATE_START_RANGE_PROPERTY] = True
                    date_dicts[1][detector_constant. DATE_END_RANGE_PROPERTY] = True
                    date_dict_list.extend(date_dicts)
```

This also applies to lines 219-224 but since that is out of the scope of this PR, I wouldn't suggest changing it right away.